dace/transformation/interstate/gpu_transform_sdfg.py

```diff
@@ -214,6 +217,26 @@ def apply(self, _, sdfg: sd.SDFG):
             name = sdfg.add_datadesc('gpu_' + onodename, newdesc, find_new_name=True)
             cloned_arrays[onodename] = name
 
+            # The following ensures that when writing to a subset of an array, we don't overwrite the rest of the array
+            # when copying back to the host. This is done by adding the array to the `inputs_nodes,` while will copy
+            # the entire array to the GPU.
+            if (onodename, onode) not in input_nodes:
+                found_full_write = False
+                full_subset = sbs.Range.from_array(onode)
+                try:
+                    for state in sdfg.nodes():
+                        for node in state.nodes():
+                            if (isinstance(node, nodes.AccessNode) and node.data == onodename):
+                                for e in state.in_edges(node):
+                                    if e.data.get_dst_subset(e, state) == full_subset:
```

Comment:
would this work for dynamic memlets (i.e., writing to one index but the subset is the full array)

Reply:
No, it will not work. Are you suggesting that we fail this check if the memlet is dynamic? Good idea.